### PR TITLE
SF-2502 Add Help Videos to Scripture Forge

### PR DESCRIPTION
**Pull request by kylebuss**
This PR is a basic proof of concept for adding a Scripture Forge Help Video page for users.

It adds a Menu Item (Tutorials) that links to the help-video page which would be where the videos are viewable.
It adds a Help Video tab to the System Administrator page. The current behavior is just showing one video with a link to the help-video page. The idea behind this would be where we would add/update the videos we are displaying.

Further discussion is needed on hosting/general process.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2897)

### CODE REVIEW

**Review by kylebuss — COMMENTED** [review]
Self hosting videos (a new role may be needed):
Pros - Upload and Data Entry in a single step, 100% control/privacy out of the box, no 3rd party API should be needed 
Cons - Potential performance/concurrency issues, storage space on server

Platform hosting videos:
Pros - Reliable performance/concurrency, saves storage space on server
Cons - Potentially requires 3rd party API for best integration, Upload video and data entry requires multiple steps, video privacy settings managed in platform (additional setup step), unknown management limitations (access to platform to upload/delete/edit).

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-:-ODl8Z2nE6vzM57CZNe9:b4kwslw)* status: 0 of 14 files reviewed, 5 unresolved discussions (waiting on @github-advanced-security[bot])

___
*[`.gitignore` line 18 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-ODl6oHQB8tWA0nvwbx0:-ODl6oHQB8tWA0nvwbx1:bxb2od4) ([raw file](https://github.com/sillsdev/web-xforge/blob/ffaedd532202c0fe6e52f6b6148dd2f047389397/.gitignore#L18)):*
> ```ignore
> dist/
> src/RealtimeServer/lib/
> #src/SIL.XForge.Scripture/ClientApp/src/assets/videos/*
> ```

For demo purposes this was commented out.

The more I think about it, this is the least viable solution to store files here. Unless Angular would be able to handle files being added via the app itself to the asset folder?



_Code quote:_
```ignore
#src/SIL.XForge.Scripture/ClientApp/src/assets/videos/*
```
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/help/help-videos/help-videos.component.html` line 3 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-ODl75ne-txzXAbhW8or:-ODl75ne-txzXAbhW8os:b-n17vni) ([raw file](https://github.com/sillsdev/web-xforge/blob/ffaedd532202c0fe6e52f6b6148dd2f047389397/src/SIL.XForge.Scripture/ClientApp/src/app/help/help-videos/help-videos.component.html#L3)):*
> ```HTML
> <div class="help-video">
>   <video width="640" height="480" controls controlslist="nodownload" oncontextmenu="return false">
>     <source src="./assets/videos/SIL Scripture Forge Tutorial.mp4" type="video/mp4" />
> ```

If we store videos similar to how we store project audio the `src=...` would be the file path to the video file on the server. We'd have to test concurrency to see if the file gets locked and this is not a feasible solution.



_Code quote:_
```HTML
<source src="./assets/videos/SIL Scripture Forge Tutorial.mp4" type="video/mp4" />
```
___
*[`src/SIL.XForge.Scripture/ClientApp/src/xforge-common/system-administration/sa-help-video-tab/sa-help-videos/sa-help-videos.component.html` line 12 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-ODl7ioU4vS4O2xH4AtJ:-ODl7ioU4vS4O2xH4AtK:b4es371) ([raw file](https://github.com/sillsdev/web-xforge/blob/ffaedd532202c0fe6e52f6b6148dd2f047389397/src/SIL.XForge.Scripture/ClientApp/src/xforge-common/system-administration/sa-help-video-tab/sa-help-videos/sa-help-videos.component.html#L12)):*
> ```HTML
>     </ng-container>
>   </table>
> </div>
> ```

This would need to be updated to have editable fields but is the main way of adding/deleting/updating a video without having to do a new push.



_Code quote:_
```HTML
<div class="help-controls">
  <table id="help-videos">
    <ng-container matColumnDef="name">
      <td>SIL Scripture Forge Tutorial</td>
    </ng-container>
    <ng-container matColumnDef="url">
      <td>
        <a appRouterLink="../help-videos">Tutorials</a>
      </td>
    </ng-container>
  </table>
</div>
```

**Review by josephmyers — COMMENTED** [review]
It seems like you have stronger feelings about this than I do, so I'm not going to try to talk you out of this way. If you've done this before at past jobs, that's great, and your experience would trump mine. But I don't reach the same conclusions you do, given what you're presenting. There are additional pro's and con's to me with each of these, e.g. writing and maintaining the solution to stand up self-hosting, which you haven't fully done here. I'm thinking of the file upload capability and UI, the extra role, and all other database interaction. It'd be nice to see this in a demo like this, and envisioning what you haven't written yet doesn't convince me this is far and away the best route. It does give us full control -- my question is, "Do we need it?"

I'm also not seeing the need (yet) for a 3rd party tool for video hosting from an existing platform, and we are not the first people to want YouTube videos on a website. It seems like iframe can be tailored to prevent users from easily downloading our videos (with `sandbox`). We were already using YouTube videos on one of our old help pages (though it may have been removed), so why do we need to reinvent the wheel here?

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-:-ODnGDd70Hs16IM4m1e8:b-fvr5kw)* status: 0 of 14 files reviewed, 5 unresolved discussions (waiting on @github-advanced-security[bot] and @kylebuss)

**Comment by kylebuss** [comment]
I tested out the `iframe` suggestion and after working through a small issue (wrong `src` from YouTube), I was pleasantly surprised. I'm not opposed to doing it this way as YouTube already prevents users from easily downloading videos.

Hopefully I can find some time to work on this a bit more to give you a clearer picture of what I have in mind.

**Review by josephmyers — COMMENTED** [review]
Sounds good :thumbsup: 

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-:-ODsI2rc5TmB8Di2UEKf:b-ux0ur8)* status: 0 of 14 files reviewed, 5 unresolved discussions (waiting on @github-advanced-security[bot] and @kylebuss)

File: src/SIL.XForge.Scripture/ClientApp/src/xforge-common/system-administration/sa-help-video-tab/sa-help-videos/sa-help-videos.component.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Component, OnInit } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (kylebuss) [reply src/SIL.XForge.Scripture/ClientApp/src/xforge-common/system-administration/sa-help-video-tab/sa-help-videos/sa-help-videos.component.ts]: Done.

File: src/SIL.XForge.Scripture/ClientApp/src/app/help/help-videos/help-videos.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Component, OnInit } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (kylebuss) [reply src/SIL.XForge.Scripture/ClientApp/src/app/help/help-videos/help-videos.component.ts]: Done.

**Review by kylebuss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2897)* status: 0 of 17 files reviewed, 3 unresolved discussions (waiting on @github-advanced-security[bot])

**Review by josephmyers — COMMENTED** [review]
Cool, I didn't think of using the manage videos workflow for the YouTube solution. I like it

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2897#-:-OEC2NMeDs_B1DVJjOup:b4h23p2)* status: 0 of 17 files reviewed, 3 unresolved discussions (waiting on @github-advanced-security[bot])

**Comment by kylebuss** [comment]
Closing this PR after discussing with Nathaniel. Updating this issue to improve integrating links to the help site across the app rather than just adding help videos (which are now on the Help site).

**Pull request closed without merging**